src/providers/BankTransactionsProvider/BankTransactionsProvider.tsx

```diff
@@ -1,18 +1,51 @@
-import { ReactNode } from 'react'
+import { ReactNode, useMemo } from 'react'
 import { BankTransactionsContext } from '../../contexts/BankTransactionsContext'
 import { useAugmentedBankTransactions } from '../../hooks/useBankTransactions/useAugmentedBankTransactions'
+import {
+  TransactionsFiltersContext,
+  useTransactionsFilters,
+  UseTransactionsFiltersParams,
+} from '../../contexts/TransactionsFiltersContext'
+import { collectAccounts } from '../../hooks/useBankTransactions/utils'
 
-interface BankTransactionsProviderProps {
+interface BankTransactionsProviderProps extends UseTransactionsFiltersParams {
   children: ReactNode
 }
 
 export const BankTransactionsProvider = ({
   children,
+  scope,
+  monthlyView,
+  applyGlobalDateRange,
 }: BankTransactionsProviderProps) => {
-  const bankTransactionsContextData = useAugmentedBankTransactions()
+  const { filters, setFilters, dateFilterMode } = useTransactionsFilters({
```

Comment:
I think once we memoize `useTransactionsFilters`, we can take the response value and directly shove it in `TransactionsFiltersContext`.
```suggestion
  const filtersContextValue = useTransactionsFilters({
```